### ISSUE TITLE
if metrics not provided, not handled in trainer. Pass random_seed to dataset and training_table

### ISSUE THREAD

**Issue by anithapk** [czbiohub/microDL#66]
callbacks = self._get_callbacks()
       if 'metrics' in self.config:
208             metrics_list = self.config['metrics']
209             metrics = get_metrics(metrics_list)
210             self._compile_model(loss, optimizer, metrics)
211         else:
212             self._compile_model(loss, optimizer, metrics=None)

`` gpu_id is an int whereas mem_frac is a list
    run_action(args, gpu_id, gpu_mem_frac[0])

if loss_is_masked:
177             if 'metrics' in self.config:
178                 masked_metrics = [metric(self.num_target_channels)
179                                   for metric in metrics]
180                 self.model.compile(loss=masked_loss(loss,
181                                                     self.num_target_channels),
182                                    optimizer=optimizer,
183                                    metrics=masked_metrics)
184             else:
185                 self.model.compile(loss=masked_loss(loss,
186                                                     self.num_target_channels),
187                                    optimizer=optimizer,
188                                    metrics=None)

Unet_stack_to_2D
@property
    def _get_input_shape(self):
        """Return shape of input"""

        if self.config['data_format'] == 'channels_first':
            shape = (1,
                     self.config['depth'],
                     self.config['height'],
                     self.config['width'])
        else:
            shape = (self.config['depth'],
                     self.config['height'],
                     self.config['width'], 1)
        return shape

num of channels is hardcoded at 1! fix it
